packages/discovery-provider/src/queries/search_es.py

```diff
@@ -386,23 +340,61 @@ def track_dsl(
 ):
     dsl = {
         "must": [
-            *base_match(search_str),
             {"term": {"is_unlisted": {"value": False}}},
             {"term": {"is_delete": False}},
+            {
+                "bool": {
+                    "should": [
+                        *base_match(search_str),
+                        {"wildcard": {"title": "*" + search_str + "*"}},
+                        {
+                            "multi_match": {
+                                "query": search_str,
+                                "fields": ["title.searchable", "user.name.searchable"],
+                                "type": "cross_fields",
+                                "operator": "and",
+                                "boost": len(search_str) * 0.5,
```

Comment:
So this helps in cases where small search terms don't over boost exact matches?

Reply:
this takes all the terms in the search_str and slightly boosts if they all appear in a title or user name (cross_fields). compared to the others which are looking for prefix matching on each field separately. boosting 0.5 doesn't penalize it still slightly boosts.